aslam_cv_pipeline/src/visual-npipeline.cc

```diff
@@ -122,66 +133,76 @@ std::shared_ptr<VisualNFrame> VisualNPipeline::getNext() {
 
 std::shared_ptr<VisualNFrame> VisualNPipeline::getNextImpl() {
   // Initialize the return value as null
-  std::shared_ptr<VisualNFrame> rval;
-  if(!completed_.empty()) {
-    // Get the oldest frame.
-    auto it = completed_.begin();
-    rval = it->second;
-    completed_.erase(it);
-    condition_not_full_.notify_all();
+  std::shared_ptr<VisualNFrame> nframe;
+  if (shutdown_ || completed_.empty()) {
+    return nframe;
   }
-  return rval;
+  // Get the oldest frame.
+  auto it = completed_.begin();
```

Comment:
no auto ;-)

Reply:
In this case I agree with the google c++ style guide that encourages the use of auto for iterators. I think it is just much more readable and also it is clear what the type is.

Reply:
oh they encourage that? didn't know that ;-) ok, fair enough...